lib/sles4sap_publiccloud.pm

```diff
@@ -182,6 +184,70 @@ sub sles4sap_cleanup {
 
 }
 
+=head2 get_job_state
+    Get whether a specified job is still running or not
+=cut
+
+sub get_job_state {
+    my ($job_id) = @_;
+
+    my $url = "https://openqa.suse.de/api/v1/jobs/$job_id";
+
+    my $json_data = script_output("curl -s '$url'");
+
+    my $job_data = eval { decode_json($json_data) };
+    if ($@) {
+        warn "Failed to decode JSON data for job $job_id: $@";
+        return 0;    # Assume job is still running if we can't get its state
+    }
+
+    my $job_state = $job_data->{job}->{state} // '';
+
+    if ($job_state eq 'running') {
```

Comment:
Not entirely sure but I think Perl support

```
return ($job_state eq 'running') ? 0 : 1;
```

Reply:
Done